### ISSUE TITLE
Can we delete a whole stream?

### ISSUE THREAD

**Issue by datafoo** [commanded/commanded#267]
I am considering using event sourcing to store Personally Identifying Information (PII), specifically user details such as username and display name.
However, if I do that, I would like to be able to delete a whole stream of events in order to comply with various international rules regarding PII.

[Commanded.EventStore](https://github.com/commanded/commanded/blob/5a19dcb14813940398212481d3600f1c278e39a6/lib/commanded/event_store/event_store.ex#L154) already provides a way to interact with the event store to append one or more events to a stream. Would that be possible/realistic to provide a function to delete a whole stream?

**Comment by slashdotdash** [comment]
Greg Young's Event Store supports [soft and hard deletion of streams](https://eventstore.org/docs/server/deleting-streams-and-events/) but the Postgres based Elixir EventStore doesn't yet support deleting streams ([#157](https://github.com/commanded/eventstore/issues/157)). 

To add this feature to Commanded requires each event store to provide deletion. I think it's definitely worth adding.

**Comment by datafoo** [comment]
> To add this feature to Commanded requires *each* event store to provide deletion.

Is there any other event store than https://github.com/commanded/eventstore that needs to have the feature added?

**Comment by slashdotdash** [comment]
Yes, the in-memory event store would also need to be implemented, but that wouldn't take much effort.

**Comment by Freyskeyd** [comment]
should we update aggregates/event handler/... when a stream is deleted?
